### PR TITLE
[MNT] Add distribution doctest coverage

### PR DESCRIPTION
**Pull request by russlan23**
#### Reference Issues/PRs

First module slice of #145.

#### What does this implement/fix? Explain your changes.

Enables pytest doctest collection for `src/skfolio/distribution` while keeping the normal `tests` suite in the default collection paths. The configuration uses `NORMALIZE_WHITESPACE` and `ELLIPSIS` as proposed in the issue.

The existing distribution examples are updated to execute cleanly: fitted estimators no longer emit an unasserted repr, displayed floating-point values use explicit precision, stale fitted representations are corrected, and the multiline `VineCopula` example uses valid doctest continuation syntax.

Sphinx Gallery, other Sphinx pages, and other skfolio modules remain outside this PR, following the gradual rollout discussed in #145.

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

- Whether `testpaths = ["tests", "src/skfolio/distribution"]` is the preferred way to preserve the normal suite while limiting the first doctest rollout.
- Whether the output assertions strike the right balance between useful examples and stability across supported dependency versions.

#### Did you add any tests for the change?

The distribution docstrings are now collected as tests.

Validation in an isolated Python 3.12 container:

- `python -m pytest -q src/skfolio/distribution` - 10 passed
- `ruff check pyproject.toml src/skfolio/distribution` - passed
- `ruff format --check pyproject.toml src/skfolio/distribution` - passed
- `git diff --check` - passed

The complete `.[dev]` environment was not available in the local ARM container because PySCIPOpt 5.7.1 requires a system SCIP/compiler installation there; the upstream test matrix can provide that solver-backed coverage.

#### Any other comments?

This is intentionally the first `skfolio/distribution` slice requested in the issue discussion.

##### For all contributions

- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].

##### For new estimators

- [ ] Not applicable; this PR does not add an estimator.

### CODE REVIEW

**Comment by HugoDelatte** [comment]
Thanks for the PR!

I think the best compromise is to treat docstring examples as lightweight smoke tests and favour readability. So we should avoid making examples harder to read only to satisfy strict doctest matching, doctest settings should be relaxed instead.

So I agree that `NORMALIZE_WHITESPACE` and `ELLIPSIS` make sense for this purpose and help keep examples readable.

I also agree with you that adding doctest under pytest is good idea: `uv run pytest` should run the doctests, so developers do not need to remember another command.

However, In CI the doctests should run only once in the representative coverage job. The other matrix jobs should restrict pytest to the unit tests:

```shell
uv run pytest -q tests
```

The representative job can run:

```shell
uv run pytest -q --cov=skfolio --cov-report=xml
```

With the configured `testpaths`, this runs both `tests/` and the distribution docstrings.


For fitting, I would follow the scikit-learn style:

```python
>>> model = Model(...)
>>> model.fit(X)
Model(...)
```

This reads better than chaining and avoids `_ = model.fit(X)`.

For numerical output, I would avoid formatting f-strings:

```python
>>> print(model.lower_tail_dependence)
0.276...
```

For the same reason: it's easier to read.


There is one issue with the plots `fig.plot()`: when we run the tests, they all display in the user browser.

So I would drop `fig.plot()` and just call:

```python
>>> fig = model.plot_pdf()
```

I would also add this to `CONTRIBUTING.md` to make it clear about readability trade-off:

```markdown
#### Docstring examples

Examples in public docstrings are run as doctests. They provide a basic check that the documented workflow works, but do not replace tests in `tests/`.

Keep examples simple and readable. Avoid code written only to satisfy doctest. Put exact numerical checks, edge cases, and regression tests in `tests/`.
```

**Comment by russlan23** [comment]
Thanks, I've incorporated the requested changes in `ecd84d13`:

- non-coverage Linux jobs and cross-OS jobs now run `uv run pytest -q tests`, while the representative coverage job keeps the full configured test paths;
- fit examples use the estimator display style instead of assigning to `_`;
- variable numerical outputs use `ELLIPSIS` rather than formatting f-strings;
- plot examples construct the figures without calling `show()`; and
- `CONTRIBUTING.md` now documents the readability and smoke-test intent of docstring examples.

Local validation in an isolated Python 3.13 container:

- `pytest -q src/skfolio/distribution`: 10 passed
- `ruff check src/skfolio/distribution`: passed
- `ruff format --check src/skfolio/distribution`: 23 files already formatted
- CI workflow YAML parsing and `git diff --check`: passed

I also ran `pytest -q tests` with the core dependencies: 4,622 tests passed. The remaining 22 were environment-limited and outside this patch (20 require SCIP/PySCIPOpt, which was unavailable in the container, and two entropy-pooling cases failed with the latest Python 3.13 dependency set). The project CI should now exercise the complete declared development environment.

**Comment by HugoDelatte** [comment]
Perfect, looks good.
I'll merge it after PR https://github.com/skfolio/skfolio/pull/253 is merged, as it solved the unrelated test failures.

**Comment by russlan23** [comment]
PR #253 has merged, so I brought current main into this branch and resolved the three conflicts while preserving the distribution-doctest scope. Local verification on Python 3.13.15: 10/10 distribution doctests passed; Ruff check passed; Ruff format check reported all 23 files already formatted; the CI YAML parsed; and the branch-only diff check passed. PR #254 is mergeable again.

**Comment by HugoDelatte** [comment]
Thanks, there is a small unrelated test failure that I will tighten.
Then it will be good to merge.

**Review by HugoDelatte — APPROVED** [review]
Looks good, thanks for the PR!
